domainlab/utils/generate_benchmark_plots.py

```diff
@@ -30,53 +33,69 @@ def gen_benchmark_plots(agg_results: str, output_dir: str):
                          converters={COLNAME_PARAM: literal_eval},
                          # literal_eval can safe evaluate python expression
                          skipinitialspace=True)
-    raw_df[COLNAME_PARAM] = round_vals_in_dict(raw_df[COLNAME_PARAM])
-    # crop param_index and task from the dataframe
-    dataframe = raw_df.iloc[:, 2:]  # @FIXME: hard coded
+    raw_df[COLNAME_PARAM] = round_vals_in_dict(raw_df[[COLNAME_IDX_PARAM, COLNAME_PARAM]],
+                                               use_param_index)
     # generating plot
-    gen_plots(dataframe, output_dir)
+    gen_plots(raw_df, output_dir, use_param_index)
 
 
-def round_vals_in_dict(df_column_in):
+def round_vals_in_dict(df_column_in, use_param_index):
     '''
     replaces the dictionary by a string containing only the significant digits of the hyperparams
-    df_column_in: columns of the dataframe containing the dictionary of hyperparams
+    or (if use_param_index = True) by the parameter index
+    df_column_in: columns of the dataframe containing the param index and the dictionary of
+        hyperparams in the form [param_index, params]
+    use_param_index: usage of param_index instead of exact values
     '''
     df_column = df_column_in.copy()
-    df_column_out = df_column_in.copy()
+    df_column_out = df_column_in.iloc[:, 0].copy()
+    df_column_out = df_column_out.astype(str)
     for i in range(df_column.shape[0]):
-        string = ''
-        for num, val in enumerate(list(df_column[i].values())):
-            key = list(df_column[i].keys())[num]
-            val = np.format_float_scientific(val, precision=1, unique=False, trim='0')
-            string += str(key) + ': ' + str(val) + ', '
-        df_column_out[i] = string[:-2]
+        if not use_param_index:
+            string = ''
+            for num, val in enumerate(list(df_column.iloc[i, 1].values())):
+                key = list(df_column.iloc[i, 1].keys())[num]
+                val = np.format_float_scientific(val, precision=1, unique=False, trim='0')
+                string += str(key) + ': ' + str(val) + ', '
+            df_column_out[i] = string[:-2]
```

Comment:
could you add some comments what is this -2?

Reply:
the string is build in the for loop above, for each hyperparameter the string will be appended by 
`param_name: param_value, `
hence in the end we always have a `, `. The slicing [:-2] does remove this `, ` in the end of the string.

I will also add a comment directly into the code to explain this step